--- before/net/minecraft/scoreboard/Scoreboard.java
+++ after/net/minecraft/scoreboard/Scoreboard.java
@@ -202,15 +202,15 @@
     {
         this.scoreObjectives.remove(objective.getName());
 
-        for (int i = 0; i < 19; i++)
+        for (int i = 0; i < 19; ++i)
         {
             if (this.getObjectiveInDisplaySlot(i) == objective)
             {
                 this.setObjectiveInDisplaySlot(i, null);
             }
         }
 
         List<ScoreObjective> list = this.scoreObjectiveCriterias.get(objective.getCriteria());
 
         if (list != null)
         {
@@ -445,7 +445,7 @@
         {
             displaySlots = new String[19];
 
-            for (int i = 0; i < 19; i++)
+            for (int i = 0; i < 19; ++i)
             {
                 displaySlots[i] = getObjectiveDisplaySlot(i);
             }
