--- before/net/minecraft/util/text/ITextComponent.java
+++ after/net/minecraft/util/text/ITextComponent.java
@@ -34,7 +34,6 @@
 
     String getUnformattedText();
 
-    @SideOnly(Side.CLIENT)
     String getFormattedText();
 
     List<ITextComponent> getSiblings();
@@ -97,7 +96,7 @@
                             JsonArray jsonarray = jsonobject.getAsJsonArray("with");
                             Object[] aobject = new Object[jsonarray.size()];
 
-                            for (int i = 0; i < aobject.length; i++)
+                            for (int i = 0; i < aobject.length; ++i)
                             {
                                 aobject[i] = this.deserialize(jsonarray.get(i), p_deserialize_2_, p_deserialize_3_);
 
@@ -116,7 +115,7 @@
                         }
                         else
                         {
-                            itextcomponent = new TextComponentTranslation(s);
+                            itextcomponent = new TextComponentTranslation(s, new Object[0]);
                         }
                     }
                     else if (jsonobject.has("score"))
@@ -158,7 +157,7 @@
                             throw new JsonParseException("Unexpected empty array of components");
                         }
 
-                        for (int j = 0; j < jsonarray2.size(); j++)
+                        for (int j = 0; j < jsonarray2.size(); ++j)
                         {
                             itextcomponent.appendSibling(this.deserialize(jsonarray2.get(j), p_deserialize_2_, p_deserialize_3_));
                         }
